ocaml/idl/datamodel_errors.ml

```diff
@@ -107,6 +107,10 @@ let _ =
     ~doc:"A VDI with the specified location already exists within the SR" ();
   error Api_errors.memory_constraint_violation ["constraint"]
     ~doc:"The dynamic memory range does not satisfy the following constraint." ();
+  error Api_errors.memory_constraint_violation_order ["constraint"]
```

Comment:
I think that the parameters need to be better defined. Right now, it is still a long string that includes the doc string below and (for maxpin) a reason appended to that string. It would be better if the error code + arguments are "machine readable", so that they can be easily internationalised etc. by a client such as XenCenter.

So, I think that the `_order` error does not need an argument, while the `maxpin` error needs a reason error that tells the client why this stricter constraint applies.